### PR TITLE
New package: yabar-0.4.0

### PR DESCRIPTION
**Pull request by subsonik**
My first submission in a while, sorry for any mistakes. :)

### CODE REVIEW

File: srcpkgs/yabar/template [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+version=0.4.0
+revision=1
+build_style=gnu-makefile
+hostmakedepends="pkg-config"
+makedepends="pango-devel xcb-util-wm-devel libconfig-devel"
+short_desc="A modern and lightweight status bar for X window managers"
+maintainer="John Rowley <enterthevoid@codesector.co>"
+license="MIT"
+homepage="https://github.com/geommer/yabar"
+distfiles="https://github.com/geommer/$pkgname/archive/$version.tar.gz"
```
Comment (Vaelatern) [inline srcpkgs/yabar/template]: Use `$homepage` here

File: srcpkgs/yabar/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+checksum=f8cc613413328af78944ccb8063a5ba8e6be11797a883c419876e973e102fed0
+nopie=yes
+
+pre_build() {
+	sed -i "s|\$(VERSION)|${version}|" Makefile
+}
+
+do_build() {
+	make ${makejobs}
```
Comment (Vaelatern) [inline srcpkgs/yabar/template]: Please add our CFLAGS and LDFLAGS, then remove `nopie=yes`

**Comment by subsonik** [comment]
Upstream Makefile has -Os hardcoded into it, is this okay or should user -O level take precedent?

**Comment by Vaelatern** [comment]
Our -O should take precedent.

File: srcpkgs/yabar/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+checksum=f8cc613413328af78944ccb8063a5ba8e6be11797a883c419876e973e102fed0
+
+pre_build() {
+	sed -ie "s|^VERSION.*\$|VERSION = ${version}|" Makefile
+	sed -ie "/^LDLIBS/s|:= \(.*\)\$|:= ${LDFLAGS} \1|" Makefile
+}
+
+do_build() {
+	CFLAGS=$CFLAGS make $makejobs
+}
```
Comment (Vaelatern) [inline srcpkgs/yabar/template]: Why do you just re-implement the `do_build` of `gnu-makefile`? Try removing this function.

Comment (subsonik) [inline srcpkgs/yabar/template]: Commenting out the do_build() function results in:

```
=> yabar-0.4.0_1: running do_build ...
cc -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2 -mtune=generic -march=native -O2 -pipe        -DVERSION=\"0.4.0\" -D_POSIX_C_SOURCE=199309L -DYA_INTERNAL -DYA_DYN_COL -DYA_ENV_VARS -DYA_INTERNAL_EWMH  -c -o src/ya_parse.o src/ya_parse.c
cc -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2 -mtune=generic -march=native -O2 -pipe        -DVERSION=\"0.4.0\" -D_POSIX_C_SOURCE=199309L -DYA_INTERNAL -DYA_DYN_COL -DYA_ENV_VARS -DYA_INTERNAL_EWMH  -c -o src/ya_exec.o src/ya_exec.c
cc -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2 -mtune=generic -march=native -O2 -pipe        -DVERSION=\"0.4.0\" -D_POSIX_C_SOURCE=199309L -DYA_INTERNAL -DYA_DYN_COL -DYA_ENV_VARS -DYA_INTERNAL_EWMH  -c -o src/ya_draw.o src/ya_draw.c
cc -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2 -mtune=generic -march=native -O2 -pipe        -DVERSION=\"0.4.0\" -D_POSIX_C_SOURCE=199309L -DYA_INTERNAL -DYA_DYN_COL -DYA_ENV_VARS -DYA_INTERNAL_EWMH  -c -o src/ya_main.o src/ya_main.c
cc -specs=/void-packages/common/environment/configure/gccspecs/hardened-cc1  -fstack-protector-strong -D_FORTIFY_SOURCE=2 -mtune=generic -march=native -O2 -pipe        -DVERSION=\"0.4.0\" -D_POSIX_C_SOURCE=199309L -DYA_INTERNAL -DYA_DYN_COL -DYA_ENV_VARS -DYA_INTERNAL_EWMH  -c -o src/intern_blks/ya_intern.o src/intern_blks/ya_intern.c
src/ya_exec.c:9:19: fatal error: yabar.h: No such file or directory
 #include "yabar.h"
                   ^
compilation terminated.
src/ya_main.c:9:19: fatal error: yabar.h: No such file or directory
 #include "yabar.h"
                   ^
compilation terminated.
src/ya_draw.c:9:19: fatal error: yabar.h: No such file or directory
 #include "yabar.h"
                   ^
compilation terminated.
make: *** [<builtin>: src/ya_draw.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [<builtin>: src/ya_exec.o] Error 1
make: *** [<builtin>: src/ya_main.o] Error 1
src/ya_parse.c:9:19: fatal error: yabar.h: No such file or directory
 #include "yabar.h"
                   ^
compilation terminated.
src/intern_blks/ya_intern.c:9:19: fatal error: yabar.h: No such file or directory
 #include "yabar.h"
                   ^
compilation terminated.
make: *** [<builtin>: src/ya_parse.o] Error 1
make: *** [<builtin>: src/intern_blks/ya_intern.o] Error 1
=> ERROR: yabar-0.4.0_1: failed to run do_build() at line 8.
```

I think it has something to do with this in the Makefile:

```
CFLAGS += -std=c99 -Iinclude -pedantic -Wall -Os `pkg-config --cflags pango pangocairo libconfig`
```

It's hardcoding -Iinclude, which is an include folder in the dist. But it's +=, so I'm not sure why locally set CFLAGS is completely replacing the Makefile's CFLAGS...

I'll have a further look into it tomorrow when I'm less tired, unless you can see something obviously wrong? Attached is the Makefile.

[Makefile.txt](https://github.com/voidlinux/void-packages/files/296658/Makefile.txt)

Comment (Vaelatern) [inline srcpkgs/yabar/template]: Try, in the base of the template, not in any functions, `export CFLAGS='-Iinclude'`

Comment (ebfe) [inline srcpkgs/yabar/template]: This is the old `gnu-makefile` runs `make CFLAGS=...` problem. IMO using a custom `do_build` is less problematic than replicating the CFLAGS from the Makefile in the template.

Comment (Vaelatern) [inline srcpkgs/yabar/template]: Alright. @yopito Just keep the `do_build()` and make sure it honours our CFLAGS (by not setting `nopie=yes`)

Comment (subsonik) [inline srcpkgs/yabar/template]: OK, thanks. I believe this should be good to go then? xbps's CFLAGS and LDFLAGS are respected, although the Makefile's take precedent. LDFLAGS are added to the Makefile's 'LDLIBS' which gets passed to the linking stage anyway.

**Comment by Vaelatern** [comment]
The rule within Void is that for every template altered, there should be one commit. Those commits should be named as directed by the end of the [Manual (link)](https://github.com/voidlinux/void-packages/blob/master/Manual.md#contributing-via-git). As far as the repository is concerned, nothing was done until the package was built for the repository, so the commits for various work done can be squashed into just one commit.

**Pull request closed without merging**